lib/power_action_utils.pm

```diff
@@ -357,7 +357,10 @@ sub power_action {
             systemctl 'poweroff';
         }
 
+        # Swicth to PVM mgmt console then video can capture some logs in case shutdown fails
+        reconnect_mgmt_console if (is_pvm && ($action eq 'poweroff'));
         assert_shutdown_with_soft_timeout($soft_fail_data) if ($action eq 'poweroff');
+
```

Comment:
there is already a function `prepare_system_shutdown`. This seems to fit much better there. Can you move it there?

Also as thoughts for future improvements: For better delegation of responsibilities it would be good to not do any "if $backend" checks in the calls but delegate to functions that do the right decision per backend. So effectively we can consider to add a backend method "prepare_for_shutdown" or "prepare_consoles_for_shutdown" and each backend can decide what is best. In case of PVM that would be to switch to the mgmt console.

Reply:
Thanks, let me try with that!

Reply:
@okurz I don't think `prepare_system_shutdown` is better place, since:

1. It is better for us to shutdown the LPAR from root console rather than `vterm` console IMO
2.  It will change the logic much since `prepare_system_shutdown` is still used by other function/method.

I agree with you in future improvements, backend can capture the `vterm` logs to a new `terminal` rather than video or a file. thanks.